apps/testingaccessibility/src/utils/send-server-email.ts

```diff
@@ -0,0 +1,120 @@
+import {NextAuthOptions} from 'next-auth'
+import {nextAuthOptions} from '../pages/api/auth/[...nextauth]'
+import {createHash, randomBytes} from 'crypto'
+
+const baseUrl = process.env.NEXTAUTH_URL || process.env.VERCEL_URL
+
+export async function sendServerEmail({
+  email,
+  callbackUrl,
+}: {
+  email: string
+  callbackUrl?: string
+}) {
+  const signInUrl = `${baseUrl}/api/auth/signin/email`
+  const url = parseUrl(`${baseUrl}/api/auth`)
+  const secret = createSecret({userOptions: nextAuthOptions, url})
+  const {cookie} = createCSRFToken({options: {secret}, isPost: false})
+  const cookieHeader = `next-auth.csrf-token=${cookie}`
+  const clientToken = await getClientCsrfToken(cookieHeader)
```

Comment:
in the api environment, specifically in a webhook for stripe, there are no cookies in the header so the `next-auth` csrf protection doesn't work, but we want to send emails in a webhook handler so all of the above effectively recreates the header that we can use to send a magic link email